### PR TITLE
universal-pidff: init at 0.0.8

### PR DESCRIPTION
**Pull request by computerdane**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by danerieber** [comment]
Result of `nixpkgs-review pr 334445` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19.universal-pidff</li>
    <li>linuxKernel.packages.linux_4_19_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_10.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_10_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_15.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_15_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_4.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_4_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_1.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_10.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_1_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_6.universal-pidff</li>
    <li>linuxKernel.packages.linux_hardened.universal-pidff (linuxKernel.packages.linux_6_6_hardened.universal-pidff)</li>
    <li>linuxKernel.packages.linux_6_8.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_8_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_9.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_9_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_latest_libre.universal-pidff</li>
    <li>linuxKernel.packages.linux_libre.universal-pidff</li>
    <li>linuxKernel.packages.linux_lqx.universal-pidff</li>
    <li>linuxKernel.packages.linux_xanmod.universal-pidff</li>
    <li>linuxKernel.packages.linux_xanmod_latest.universal-pidff (linuxKernel.packages.linux_xanmod_stable.universal-pidff)</li>
    <li>linuxKernel.packages.linux_zen.universal-pidff</li>
  </ul>
</details>

**Review by DaRacci — APPROVED** [review]
Looks good to me!

**Comment by DaRacci** [comment]
@danerieber I've updated the package to 0.0.8, can you update the pr title.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19.universal-pidff</li>
    <li>linuxKernel.packages.linux_4_19_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_10.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_10_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_15.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_15_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_4.universal-pidff</li>
    <li>linuxKernel.packages.linux_5_4_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_1.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_10.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_1_hardened.universal-pidff</li>
    <li>linuxKernel.packages.linux_6_6.universal-pidff</li>
    <li>linuxKernel.packages.linux_hardened.universal-pidff (linuxKernel.packages.linux_6_6_hardened.universal-pidff)</li>
    <li>linuxKernel.packages.linux_latest_libre.universal-pidff</li>
    <li>linuxKernel.packages.linux_libre.universal-pidff</li>
    <li>linuxKernel.packages.linux_lqx.universal-pidff</li>
    <li>linuxKernel.packages.linux_xanmod.universal-pidff</li>
    <li>linuxKernel.packages.linux_xanmod_latest.universal-pidff (linuxKernel.packages.linux_xanmod_stable.universal-pidff)</li>
    <li>linuxKernel.packages.linux_zen.universal-pidff</li>
  </ul>
</details>

**Review by DaRacci — APPROVED** [review]

File: pkgs/os-specific/linux/universal-pidff/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+stdenv.mkDerivation rec {
+  name = "universal-pidff-${version}-${kernel.version}";
+  version = "0.0.8";
+
+  src = fetchFromGitHub {
+    owner = "JacKeTUs";
+    repo = "universal-pidff";
+    rev = "tags/${version}";
+    sha256 = "sha256-DptLqyjfRCpkbyFAQI8sylyKu2XyZZGrSITva9vcXGQ=";
```
Comment (Aleksanaa) [inline pkgs/os-specific/linux/universal-pidff/default.nix:16]:
```suggestion
    hash = "sha256-DptLqyjfRCpkbyFAQI8sylyKu2XyZZGrSITva9vcXGQ=";
```

File: pkgs/os-specific/linux/universal-pidff/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  makeFlags = [
+    "KVERSION=${kernel.modDirVersion}"
+    "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALL_MOD_PATH=$(out)"
+  ];
+  installTargets = [ "install" ];
+
+  meta = with lib; {
+    description = "Linux PIDFF driver with useful patches for initialization of FFB devices. Primarily targeting Direct Drive wheelbases.";
```
Comment (Aleksanaa) [inline pkgs/os-specific/linux/universal-pidff/default.nix:37]:
```suggestion
    description = "PIDFF driver with useful patches for initialization of FFB devices";
```

File: pkgs/os-specific/linux/universal-pidff/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    homepage = "https://github.com/JacKeTUs/universal-pidff";
+    license = licenses.gpl2;
```
Comment (Aleksanaa) [inline pkgs/os-specific/linux/universal-pidff/default.nix:39]:
```suggestion
    license = licenses.gpl2Plus;
```

File: pkgs/os-specific/linux/universal-pidff/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  makeFlags = [
+    "KVERSION=${kernel.modDirVersion}"
+    "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALL_MOD_PATH=$(out)"
+  ];
+  installTargets = [ "install" ];
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/os-specific/linux/universal-pidff/default.nix:36]:
```suggestion
  meta = {
```

File: pkgs/os-specific/linux/universal-pidff/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  kernel,
+}:
+
+stdenv.mkDerivation rec {
+  name = "universal-pidff-${version}-${kernel.version}";
```
Comment (Aleksanaa) [inline pkgs/os-specific/linux/universal-pidff/default.nix:9]:
```suggestion
  pname = "universal-pidff";
```

**Review by DaRacci — COMMENTED** [review]
@Aleksanaa I've applied the suggestions.

With removing with lib; is prefixing all the items with "lib." the way that is accepted for packages?

**Review by DaRacci — APPROVED** [review]

**Comment by DaRacci** [comment]
Finally received my Moza r5, testing and fully working.

**Comment by Aleksanaa** [comment]
@ofborg build linuxPackages.universal-pidff

**Comment by danerieber** [comment]
@Aleksanaa Thanks for the review! I also updated the Nix code on the [wiki page](https://nixos.wiki/wiki/Linux_kernel#Packaging_out-of-tree_kernel_modules) that I used as a template for this package. It now reflects the suggestions you made in this PR!

**Pull request merged**